integration_tests/run_pyspark_from_build.sh

```diff
@@ -117,10 +117,17 @@ else
     NUM_LOCAL_EXECS=${NUM_LOCAL_EXECS:-0}
     MB_PER_EXEC=${MB_PER_EXEC:-1024}
     CORES_PER_EXEC=${CORES_PER_EXEC:-1}
+    # Spark 3.1.1 includes https://github.com/apache/spark/pull/31540
+    # which helps with spurious task failures as observed in our tests. If you are running
+    # Spark versions before 3.1.1, this sets the spark.max.taskFailures to 4 to allow for
+    # more lineant configuration, else it will set them to 1 as spurious task failures are not expected
+    # for Spark 3.1.1+
+    VERSION_STRING=`$SPARK_HOME/bin/pyspark --version 2>&1|awk '/version\ [0-9.]+$/{print $NF}'`
```

Comment:
This fails for version strings with alphabetical characters in it, e.g.: `3.2.0-SNAPSHOT`

Reply:
nice catch, I'll remove the EOL check.